### PR TITLE
flow: aggregates uses fixed time slice

### PR DESCRIPTION
**Pull request by safchain**
Aggregates step now uses fixed time slice
which aims to return more stable values
across the time. By default a slice is
30 sec. but can be set as parameter
of the step.

### CODE REVIEW

**Comment by safchain** [comment]
rerun functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
rerun selenium-tests

**Comment by nplanel** [comment]
rerun selenium-tests

**Comment by nplanel** [comment]
run skydive-selenium-tests

**Comment by nplanel** [comment]
run tests

**Comment by nplanel** [comment]
run tests

**Comment by nplanel** [comment]
run tests

File: gremlin/traversal/metrics.go [hunk 1]
```diff
@@ -142,61 +146,104 @@ func (m *MetricsTraversalStep) Sum(keys ...interface{}) *traversal.GraphTraversa
...
 	}
 
-	var aggregated []common.Metric
+	context := m.GraphTraversal.Graph.GetContext()
+
+	start := context.TimeSlice.Start
+	last := context.TimeSlice.Last
+
+	steps := (last - start) / sliceLength
+	if last-start%sliceLength != 0 {
```
Comment (lebauce) [inline gremlin/traversal/metrics.go]: (last-start)%sliceLength ?

File: gremlin/traversal/metrics.go [hunk 2]
```diff
@@ -142,61 +146,104 @@ func (m *MetricsTraversalStep) Sum(keys ...interface{}) *traversal.GraphTraversa
...
 	if m.error != nil {
-		return m
+		return &MetricsTraversalStep{error: m.error}
+	}
+
+	sliceLength := defaultAggregatesSliceLength
+	if len(s) != 0 {
+		sl, ok := s[0].(int64)
+		if !ok || sl <= 0 {
+			return &MetricsTraversalStep{error: fmt.Errorf("Aggregates parameter has to be a positiv number")}
```
Comment (lebauce) [inline gremlin/traversal/metrics.go]: positive

File: gremlin/traversal/metrics.go (hunk 1 above)
Comment (nplanel) [inline gremlin/traversal/metrics.go]: if (last-start)%sliceLength != 0 {

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
rerun selenium-tests

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
rerun functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
rerun selenium-tests

**Comment by safchain** [comment]
rerun coverage

**Comment by safchain** [comment]
rerun selenium-tests

**Comment by safchain** [comment]
rerun selenium-tests

**Comment by safchain** [comment]
rerun functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
rerun functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
rerun functional-tests-backend-elasticsearch

**Pull request merged**